internal/storage/raft/backend.go

```diff
@@ -239,8 +258,18 @@ func (b *Backend) Apply(buf []byte, idx uint64) any {
 		oldVsn := res.Version
 		res.Version = strconv.Itoa(int(idx))
 
-		if err := b.store.WriteCAS(res, oldVsn); err != nil {
-			return err
+		if isRetiredType(res.GetId().GetType()) {
```

Comment:
Good point, but I don't think it practically matters.

I was assuming that a type would be "retired" when it was absolutely done and there is no way to actually access the data directly anymore. That is the case with these types as their proto definitions are gone and deregistered from the type registry.

In that case it doesn't matter if one non-upgraded server and one upgraded server differ on the database contents because there's no outward facing way to observe this difference.